broker/src/serve_tasks.rs

```diff
@@ -31,6 +32,7 @@ pub(crate) fn router() -> Router {
     Router::new()
         .route("/v1/tasks", get(get_tasks).post(post_task))
         .route("/v1/tasks/:task_id/results", get(get_results_for_task))
+        .route("/v1/tasks/:task_id/results/stream", get(get_results_for_task_stream))
```

Comment:
This might be solved more generic, as discussed in person. If the return type is still a problem (e.g. because the generic nature of `Response`exposes the content as a different type), maybe encapsulating the different Responses in an enum with appropriate `IntoResponse` is viable?